### ISSUE TITLE
Generic.Arrays.ArrayIndent with multiline item

### ISSUE THREAD

**Issue by VasekPurchart** [squizlabs/PHP_CodeSniffer#1791]
The new `Generic.Arrays.ArrayIndent` is awesome, thanks for that, but when testing adding this I encountered something I vaguely remember we were already discussing somewhere here about function calls with nested function calls, but cannot pinpoint the exact issue.

The problem is that right now, I think, the fact whether an array is multiline is simply checked as if the opening and closing brackets are on the same line or not.

But this does not work well when you have an item, that spans multiple lines itself (like a function call):

```php
<?php

$x = [foo(
	1,
	2
)];
```

This reports:

```
 6 | ERROR | [x] Closing brace of array declaration must be on a new line (Generic.Arrays.ArrayIndent.CloseBraceNotNewLine)
```

But I think this should not actually be counted as multiline array, because only the item is multiline, not the array (for example you don't wan to enforce indentation of items, you would not want to have a trailing comma behind this item etc.).

For me this is (from the perspective of the array) exactly the same as if it was written like this:

```php
<?php

$x = [foo(1, 2)];
```

This can happen with a lot of stuff, not only function calls, for example too long strings separated over multiple lines, indented expressions like ternaries etc.

So I think the condition should rather be whether start of one of the items is indented, then all of them should and the closing brace should be on new line.

**Comment by gsherwood** [comment]
You may be remembering a discussion about PSR2 multi-line arguments: http://www.php-fig.org/psr/psr-2/meta/#31---multi-line-arguments-09082013

For arrays, I've always defined a multi-line array as one where the braces are not on the same line because I apply different rules to them. So it's possible you just didn't have any checks in this area before.

I much prefer this formatting of your array, but it breaks another part of my standard that I'll need to change:
```php
$x = [
    foo(
        1,
        2
    )
];
```
But then again, I never end up writing code like this for whatever reason. I guess I might write it like this:
```php
$foo = foo(
    1,
    2
)

$x = [$foo];
```
So a change to how multi-line arrays are classified may not affect me much, but I'm not sure about everyone else. I think it's pretty hard to justify that "multi-line" doesn't mean "the array definition spans multiple lines" and I don't think I could defend that if someone came asking about it. It's why I was also against the PSR2 multi-line change - I think the wording was very clear, but it just wasn't the intention of the authors.

So after all that, maybe this needs to be an option to the sniff because I don't think a single way is going to please everyone.
